src/main/flight/pid.c

```diff
@@ -865,16 +865,9 @@ void FAST_CODE pidController(const pidProfile_t *pidProfile, timeUs_t currentTim
     float gyroRateDterm[XYZ_AXIS_COUNT];
     for (int axis = FD_ROLL; axis <= FD_YAW; ++axis) {
         gyroRateDterm[axis] = gyro.gyroADCf[axis];
-        // -----calculate raw, unfiltered D component
-
-        // Divide rate change by dT to get differential (ie dr/dt).
-        // dT is fixed and calculated from the target PID loop time
-        // This is done to avoid DTerm spikes that occur with dynamically
-        // calculated deltaT whenever another task causes the PID
-        // loop execution to be delayed.
 
         // Log the unfiltered D for ROLL and PITCH
-        if (axis != FD_YAW) {
+        if (debugMode == DEBUG_D_LPF && axis != FD_YAW) {
```

Comment:
The following maths to calculate delta is non-trivial.  It's only purpose is to provide a non-filtered D for the `D_LPF` debug, which is rarely used, except for testing the D lowpass filters.  There is no point to do all this maths unless the debug mode is `DEBUG_D_LPF`

Reply:
i saw this line of code a few weeks back.  i was wondering that since Yaw could technically have D set, why skip yaw?

Reply:
Good catch.
I also wondered about this.
The `DEBUG_D_LPF` debug also tracks pre-TPA D around line 1060, with some funky coding to determine what it is recording.
I have a feeling that the intent was to capture only pitch and roll, at the point around 877, and then likewise get pitch and roll later on after smoothing around 1060.
I'd have to check that out by seeing what goes into the debug.
The change here isn't to intended to change the functionality, just to stop doing unnecessary maths.

Reply:
OK I did a quick test of that debug.  It records pitch and roll pre-TPA D values before and after filtering.  The absolute values in the traces are not comparable, but the smoothing effect and lag are easily seen.
![Screen Shot 2023-04-25 at 12 07 06](https://user-images.githubusercontent.com/11737748/234156788-2a5da46c-7efe-4546-8d72-1dbdd4508e5a.jpg)